app/src/main/java/se/rebeccazadig/bokholken/login/UserViewModel.kt

```diff
@@ -1,44 +1,84 @@
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
+import android.database.Observable
+import android.util.Log
+import androidx.lifecycle.*
 import kotlinx.coroutines.launch
 import se.rebeccazadig.bokholken.myAdverts.UiState
 
-// internal data class UiState(
-//    /*uiState är MLD är observed i frag men all ändring sker i viewmodel*/
-//    val isDeleted: Boolean,
-//    val message: String?,
-// )
+internal data class UiStateSave(
+
+    val message: String?,
+)
+
 class UserViewModel : ViewModel() {
 
     private val loginRepo = LoginRepository.getInstance()
     private val userRepo = UserRepository.getInstance()
+    val userName = MutableLiveData("")
+    val userContact = MutableLiveData("")
+    val userCity = MutableLiveData("")
 
     val inProgress = MutableLiveData(false)
     private val _uiState = MutableLiveData(UiState(false, null))
     internal val uiState: LiveData<UiState> get() = _uiState
+    private val _uiStateSave = MutableLiveData(UiStateSave(null))
+    internal val uiStateSave: LiveData<UiStateSave> get() = _uiStateSave
 
     fun logOutInVm() {
         loginRepo.logOutInRepo()
     }
 
+    fun saveUser() {
+        inProgress.value = true
+
+        viewModelScope.launch {
+            val userid = loginRepo.getUserId()
+            val userInfo = userName.value.orEmpty()
+            val userContact = userContact.value.orEmpty()
+            val userCity = userCity.value.orEmpty()
+            val user = User(id = userid, name = userInfo, contact = userContact, city = userCity)
+
+            userRepo.saveUser(user)
+            val result = userRepo.saveUser(user)
+
+            inProgress.postValue(false)
+
+            when (result) {
+                is Result.Failure -> {
+                    Log.i("Emma", "FAIL")
+                    _uiStateSave.value = UiStateSave(message = "Hoppsan något gick fel")
+                }
+                is Result.Success -> {
+                    Log.i("Emma", "SUCCESS")
+                    _uiStateSave.value = UiStateSave(message = "Informationen Sparad")
+                }
+            }
+        }
+    }
+
     fun deleteAccountInVM() {
         inProgress.value = true
         viewModelScope.launch {
-            val deleteResult: AuthResult = loginRepo.deleteAccount()
+            val deleteResult: Result = loginRepo.deleteAccount()
             inProgress.postValue(false)
 
             when (deleteResult) {
-                is AuthResult.Failure -> {
+                is Result.Failure -> {
                     _uiState.value = UiState(false, deleteResult.message)
                 }
-                is AuthResult.Success -> {
+                is Result.Success -> {
                     _uiState.value = UiState(true, null)
                 }
             }
         }
     }
+    val isButtonDisabled = MediatorLiveData<Boolean>().apply {
```

Comment:
properties usually declared before methods

Reply:
I don't understand what you mean my mistake is

Reply:
It's not a mistake. There are the code conventions. Normally, inside a class one puts val/var lines first, then init (when needed), after that functions.